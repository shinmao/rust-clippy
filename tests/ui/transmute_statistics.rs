#![allow(unused)]
#![warn(clippy::transmute_statistics)]

pub struct struct_a {
    a: i8,
    b: i32,
    c: i8,
}

pub struct struct_b {
    a: i8,
    b: i32,
    c: i8,
}

fn main() {
    let a = 0u32;
    let a = &a as *const u32;
    let _: &u32 = unsafe { std::mem::transmute(a) };
    let _: &u32 = unsafe { std::mem::transmute::<_, &u32>(a) };

    let b: u8 = 10;
    let _ = unsafe { std::mem::transmute::<_, &u32>(&b) };

    let c: u32 = 10;
    let _ = unsafe { std::mem::transmute::<_, &u8>(&c) };

    let sa = struct_a { a: 10, b: 11, c: 12 };
    let sb: &struct_b = unsafe { std::mem::transmute(&sa) };
}
