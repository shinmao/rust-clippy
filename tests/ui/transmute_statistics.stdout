transmute caller:sa_to_sb
transmute:(&struct>&struct>warn(adt))
transmute caller:from
transmute:(struct>struct>safe)
transmute caller:main
transmute:(*const uint>&uint>safe)
transmute caller:main
transmute:(*const uint>&uint>safe)
transmute caller:main
transmute:(&uint>&uint>warn(align1>4))
transmute caller:main
transmute:(&uint>&uint>safe)
